client/lxd.go

```diff
@@ -382,11 +382,16 @@ func (r *ProtocolLXD) queryStruct(method string, path string, data any, ETag str
 
 // queryOperation sends a query to the LXD server and then converts the response metadata into an Operation object.
 // It sets up an early event listener, performs the query, processes the response, and manages the lifecycle of the event listener.
-func (r *ProtocolLXD) queryOperation(method string, path string, data any, ETag string) (Operation, string, error) {
+func (r *ProtocolLXD) queryOperation(method string, path string, data any, ETag string, useEventListener bool) (Operation, string, error) {
+	var err error
+
 	// Attempt to setup an early event listener
-	listener, err := r.GetEvents()
-	if err != nil {
-		listener = nil
+	var listener *EventListener
+	if useEventListener {
+		listener, err = r.GetEvents()
+		if err != nil {
```

Comment:
Do we need this part? Does r.GetEvents every return a non-nil listener with a non-nil error?

Reply:
Hmmm no, we can just ignore the error.

Reply:
This has just reminded me of https://github.com/canonical/lxd/issues/12011. I'm going to close it because it is a tremendous amount of work for not a great deal of pay off.